services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -64,25 +64,79 @@ class SwitchRulesSpec extends BaseSpecification {
         }
     }
 
-    @Ignore("Test is skipped because of the issue #1464")
-    def "Pre-installed rules are not deleted from a new switch when connecting it to the controller"() {
-        // TODO(ylobankov): Also the current disconnection of the switch from the controller is not suitable for
-        // this scenario. It is needed to add additional functionality to properly disconnect the switch.
+    def "Pre-installed rules (including default) are not deleted from a new switch connected to the controller"() {
+        requireProfiles("virtual")
+
+        given: "A switch with some rules installed (including default) and not connected to the controller"
```

Comment:
It'd be nice to split the given clause in two:
1. given: A switch with default rules and flows exists

2. and: Switch gets disconnected from controller

This will simplify failure analysis a bit.

Reply:
This does not influence the failure output. At current implementation Spock is not reporting which *block* has failed (there is an approved feature request though). Currently these block are for documentation purpose. As for me, the tests precondition is a "Disconnected switch pre-installed rules" and not the "Switch with flow" + "Switch gets disconnected"